pkg/csi/cinder/controllerserver.go

```diff
@@ -76,16 +76,17 @@ func (cs *controllerServer) CreateVolume(ctx context.Context, req *csi.CreateVol
 	// Required, incase vol AZ is different from node AZ
 	volAvailability = req.GetParameters()["availability"]
 
+	cloud := cs.Cloud
+	ignoreVolumeAZ := cloud.GetBlockStorageOpts().IgnoreVolumeAZ
+
 	if len(volAvailability) == 0 {
-		// Check from Topology
-		if req.GetAccessibilityRequirements() != nil {
+		// Check from Topology, if ignore-volume-az=true is set in cloud config
+		// then the AZ of compute will not be used, see #1735 for further info
+		if req.GetAccessibilityRequirements() != nil && ignoreVolumeAZ {
```

Comment:
The problem is that `req.GetAccessibilityRequirements()` is never `nil`. This is because external-provisioner is run with `--immediate-topology=true` by default. See docs for it in https://github.com/kubernetes-csi/external-provisioner#other-recognized-arguments . So if compute and volume AZs are distinct, setting `ignore-volume-az=true` still won't help. It would still use compute AZ as a volume AZ -- which is the original problem described in https://github.com/kubernetes/cloud-provider-openstack/issues/1735.

We might need an option to set `--immediate-topology=false` via Helm values, or a new cloud config option that disables  setting compute topology as a volume AZ.

Reply:
um... I need more time to read all this and maybe do some test on

for now ,looks like a `--immediate-topology=false` seems more reasonable as it's just configuration change 
will post more analysis later on

Reply:
I can share my notes I made before filing the bug report, if it helps any...

external-provisioner configuration:
* cinder-csi runs with `--immediate-topology=true`, and `--strict-topology=false`
* `--strict-topology=false` makes `CreateVolumeRequest.accessibility_requirements` = selected node
* `--immediate-topology=true` makes `CreateVolumeRequest.accessibility_requirements` = aggregated cluster topology, if `allowedTopologies` is empty
* [This table](https://github.com/kubernetes-csi/external-provisioner#topology-support) shows the different behaviors for different combinations of switches and storage class parameters
* With these options, the table linked above reduces to:
  | Delayed binding | Allowed topologies | Resulting accessibility requirements |
  |:-:|:-:|-|
  |Yes|No|`Requisite` = Aggregated cluster topology<br>`Preferred` = `Requisite` with selected node topology as first element|
  |Yes|Yes|`Requisite` = Allowed topologies<br>`Preferred` = `Requisite` with selected node topology as first element|
  |No|No|`Requisite` = Aggregated cluster topology<br>`Preferred` = `Requisite` with randomly selected node topology as first element|
  |No|Yes|`Requisite` = Allowed topologies<br>`Preferred` = `Requisite` with randomly selected node topology as first element|

cinder-csi configuration:
* `availability: <volume AZ name>` storage class parameter:
  * Not empty: `volume.Availability = parameters["availability"]`
  * Empty: `volume.Availability = head(req.AccessibilityRequirements.Preferred)`
* `ignore-volume-az=<bool>` cloud config parameter:
  * True:
  ```go
  resp.AccessibleTopologies =
    req.AccessibilityRequirements.Preferred
  ```
  * False:
  ```go
  resp.AccessibleTopologies = map[string]string{
    "topology.cinder.csi.openstack.org/zone": volume.Availability
  }
  ```

Reply:
Maybe @lingxiankong @ramineni have other ideas?

Reply:
Essentially, the issue we've run into is that `volume.Availability` is never empty: it's either what you explicitly specify the zone to be in the storage class, or external-provisioner chooses one for you -- from the compute zones.

That means Cinder can never round-robin across AZs allowed in Cinder's volume type. For clouds where compute and volume AZs don't have any relation this is a problem, because we would have to have a separate storage class for each volume AZ (where we'd have to set the `availability` param explicitly).

Reply:
And secondly, going with the assumption that "compute and volume AZs don't have any relation" is true for certain clouds, tagging the PV with anything but `nil` in `resp.AccessibleTopologies` may also bring problems (e.g. if a new compute AZ was added in the future).

IMHO the least intrusive way to handle this would be to expose `--immediate-topology` in Helm values, leaving it to `=true` by default.

Reply:
>>>IMHO the least intrusive way to handle this would be to expose --immediate-topology in Helm values, leaving it to =true by default.

I agree with above, maybe some doc (consolidate your great analysis above) should also be helpful 

@ramineni @lingxiankong thoughts?

Reply:
@jichenjc I failed to notice that cinder-csi's Helm chart already provides a way to disable topology altogether using `csi.provisioner.topology`, which is then fed into

https://github.com/kubernetes/cloud-provider-openstack/blob/ee4ecca326219153805b09ccc95911efd93f58f0/charts/cinder-csi-plugin/templates/controllerplugin-deployment.yaml#L41-L51

big apologies for all the noise and for wasting your time with this! I assumed there is only `ignore-volume-az` cloud config option, and that there is no way to disable topology.

With `csi.provisioner.topology=false` and `ignore-volume-az=true`, it seems to work just fine:
* `req.AccessibilityRequirements` is now always `nil`
* `volume.Availability` is now always `parameters["availability"]` or empty

Reply:
no worry, learn a lot from this :) thanks for reporting, I am going to close this PR